distributions/demo-server/build.gradle

```diff
@@ -96,9 +108,84 @@ shadowJar {
 
 shadowJar.dependsOn(createSecretsFile)
 
-def copyResources() {
-    copy {
-        from 'resources/'
-        into 'src/main/resources'
+task createApiServerDockerfile(type: Dockerfile) {
+    description = 'Builds the API Server Dockerfile'
+    group = 'docker'
+    destFile project.file("${buildDir}/apiserver/Dockerfile")
+    from "${javaDockerContainer}"
+    exposePort 8080
+    copyFile("build/libs/demo-server-all.jar", "/app/demo-server-all.jar")
+    defaultCommand("java", "-jar", "/app/demo-server-all.jar")
+
+    // Read in secrets and set them as environment variables.
+    // NB: This is for demo purposes only. Production systems should manage and set secrets in a secure way.
+    def secretsFile = file(Paths.get(System.getProperty('user.home')).resolve('.gradle/dataportability.secrets.properties'))
+    if (secretsFile.exists()) {
+        def props = new Properties()
+        secretsFile.withInputStream { props.load(it) }
+        props.each { k, v -> environmentVariable(k, v) }
     }
+
+}
+
+task createClientDockerfile(type: Dockerfile) {
```

Comment:
Just for clarification, this means we're running nginx within the docker container to serve the static client files after they're generated?

Reply:
Yeah another thing I was going to go into detail on. This is just for the demo. 

NGINX serves the webapp content as well as proxies to the API server in the other container. This requires a Docker network bridge to be created and the two docker images be bound to the network. One problem I have atm is if the client image is booted prior to the apiserver, NGINX throws an error as it attempts to contact the other container during initialization. I need to look into having that done lazily.

With the Docker plugin we can also have the build create the network bridge and the containers launched locally (or probably even in the cloud). We can look to do that once the basics are in place.

Reply:
Feel free to package the client as you see fit for demo purposes. As long as we can deploy the API without the client as well.  The initial thought is that the client would not be part of a deployment, as most use cases for hosting for DTP will be done behind an existing provider's UI.